src/main/scala/sparkz/core/network/peer/InMemoryPeerDatabase.scala

```diff
@@ -103,11 +103,14 @@ final class InMemoryPeerDatabase(sparkzSettings: SparkzSettings, sparkzContext:
     bucketManager.removePeer(address)
   }
 
-  override def allPeers: Map[InetSocketAddress, PeerDatabaseValue] = knownPeers ++ bucketManager.getTriedPeers ++ bucketManager.getNewPeers
+  override def allPeers: Map[InetSocketAddress, PeerDatabaseValue] =
+    if (settings.onlyConnectToKnownPeers)
+      knownPeers
+    else
+      knownPeers ++ bucketManager.getTriedPeers ++ bucketManager.getNewPeers
 
   override def blacklistedPeers: Seq[InetAddress] = blacklist
-    .map { case (address, bannedTill) =>
-      checkBanned(address, bannedTill)
+    .map { case (address, bannedTill) if checkBanned(address, bannedTill) =>
       address
     }
```

Comment:
this is not related, but important to fix.
`checkBanned(address, bannedTill)` call has a side-effect of removing peers that have served their punishments from the blacklist map. But it did not remove them from the current Seq, and this is incorrect